phoenicis-containers/src/main/java/org/phoenicis/containers/dto/ContainerCategoryDTO.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2015-2017 PÂRIS Quentin
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.phoenicis.containers.dto;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.apache.commons.lang.builder.ToStringBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Represents a category of container
+ */
+@JsonDeserialize(builder = ContainerCategoryDTO.Builder.class)
+public class ContainerCategoryDTO {
+    private final String name;
+    private final String description;
+    private final List<ContainerDTO> containers;
+    private URI icon;
+
+    private ContainerCategoryDTO(Builder builder) {
+        this.name = builder.name;
+        this.description = builder.description;
+        this.containers = Collections.unmodifiableList(builder.containers);
+        this.icon = builder.icon;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public List<ContainerDTO> getContainers() {
+        return containers;
+    }
+
+    public URI getIcon() {
+        return icon;
+    }
+
+    public static Comparator<ContainerDTO> nameComparator() {
+        return (o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName());
```

Comment:
We could do this by using `Comparator.comparing(container -> container.getName().toLowerCase())`, right?
I'm not sure which approach is more elegant.